AustinKellar/PowerOn-Prototype
# Sap System

Goal: Create a resource that controls branch growth called sap. When sap is available, branches can be grown. The platforming character can collect sap (yellow cube) in order to replenish a percentage of the sap.

Changes:
- BranchSpawner script has a new property called Sap, which tracks the amount of sap currently available
- Branches can only be grown when Sap > 0
- The maxSap value can be changed in Unity on the BranchSpawner object (starting value is 100)
- A sap bar has been added to the UI in the top right corner. It reflects the Sap variable in the BranchSpawner script. When Sap is 100, the bar is full and when Sap is 0, it is empty
- A Sap prefab has been added. It is a yellow cube (shape can be changed as needed) that when collected by the platforming player, replenishes a percentage of the maxSap. For now, I have it restoring 30% of the maxSap value. The restoration percentage can be changed in Unity in the Sap prefab.
- For testing purposes, I slowed down the growth of the branches

How to test:
- Grow branches, check if sap bar shrinks as they grow
- Increase the maxSap in the BranchSpawner object, then start the game and see if the sap bar decreases more slowly
- Collide with the yellow sap cubes. Ensure 30% of the sap bar is refilled on collection and that the sap object is destroyed immediately
- Change the refillPercentage in the Sap prefab to 50% and make sure that more sap is restored when you collect it.
